src/main.rs

```diff
@@ -156,7 +156,16 @@ fn main() {
         _ => bad_path(input),
     };
 
+    let mut exit_code = 0;
+
     while let Some(message) = messages.recv().expect("Problem receiving message") {
+        let RobloxMessage::Output { level, body: _ } = message;
+        if let OutputLevel::Error = level {
+            exit_code = 1;
+        }
```

Comment:
This is an interesting solution. I think it might be a little more Rust-ish to write this block like this:

```suggestion
        match message {
            RobloxMessage::Output { level: OutputLevel::Error, .. } => {
                exit_code = 1;
            }
        }
```

That way, if there's a new kind of `RobloxMessage`, this code does not need to change as much. We'll add a catch-all condition that does nothing and move on with our lives.

Alternatively, we could continue destructuring `message` like you did, but it might make sense to just compare `level` normally:

```rust
let RobloxMessage::Output { level, .. } = message;
if level == OutputLevel::Error {
    exit_code = 1;
}
```

We'll have to derive `PartialEq` on `OutputLevel` if it isn't already comparable.